### PR TITLE
Stop checking status of a shpec file

### PR DESCRIPTION
**Pull request by hlangeveld**
We ignore the check on the failures counter, as it's result
was never checked anywhere.

Instead, we check for the indent level going below 0,
and bail out if it does.

There is a slightly different version with  `>& 2` at the end of the line, as branch issue#65b

I prefer this one, where the redirection is next to the `printf`

### CODE REVIEW

**Comment by rylnd** [comment]
@hlangeveld awesome. I added some test coverage for this in 800e69df76b88dad5abbcd617309b3f3dbbfcab6, would you mind adding that to your branch?

**Comment by hlangeveld** [comment]
I can pull that in.

File: bin/shpec [hunk 1]
```diff
@@ -21,9 +21,9 @@ describe() {
 
 end() {
   : $((test_indent -= 1))
-  if [ $test_indent -eq 0 ]; then
-    [ $failures -eq 0 ]
-  fi
+  [ $test_indent -lt 0 ] || return 0
```
Comment (rylnd) [inline bin/shpec]: This might be easier understood as
```bash
[ $test_indent -ge 0 ] && return 0
```

**Comment by rylnd** [comment]
@hlangeveld if you can pull in those tests and rebase this branch, I'm happy to merge this.

**Comment by hlangeveld** [comment]
@rylnd Pulled in latest master and your tests (but I relaxed them a little).

**Pull request merged**